profiling/src/profile/pprof/sliced_proto.rs

```diff
@@ -0,0 +1,115 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+//! This file is a companion to the standard pprof protobuf definition.
+//! `Profile` fields are "sliced" into separate messages, allowing them to be
+//! serialized in a streamed manner.  This takes advantage of the fact that
+//! the top level message in a protobuf doesn't have a length header, so the
+//! bits on the wire are indistinguishable between serializing these sliced
+//! messages, and serializing the top-level message.
+//!
+//! The `tag` number and type for each of these sliced messages matches the
+//! corresponding field in the `Profile` message in `profile.proto`.
+
+use super::*;
+
+#[derive(Eq, Hash, PartialEq, ::prost::Message)]
+pub struct ProfileSampleTypesEntry {
+    #[prost(message, tag = "1")]
+    pub sample_types_entry: Option<ValueType>,
```

Comment:
I am confused here. Why are so many of these sliced proto fields using `Option` aka 0 or 1 instead of of a type which is 0 or more? We definitely have more than one sample type, for instance. What am I missing?

Reply:
Updated with comments in the file